proposals/0000-accept-prs-on-github.rst

```diff
@@ -0,0 +1,193 @@
+non-trivial. One needs to,
+
+* setup an account on Phabricator,
+* setup an account on Trac,
+* add a new public ssh key in Phabricator,
+* Install PHP then checkout the Git repository for arcanist to get the
+  CLI tool for code reviews going,
+* Install a new certificate in Arcanist, using ``arc install-certificate``,
+* submit one or several code reviews on Phabricator using ``arc`` commands,
+* update the Trac ticket.
+
+Many potential would-be contributors don't have unwieldy patches
+including large refactorings to the type inference engine or spiffy
+optimizations in the RTS to propose. Many of them also don't
+contribute every day, so they have to relearn the process every time.
+Whereas there is a process that many such contributors (as evidenced
+by the constant clamour) *do* use nearly every day, and which is
+particularly simple:
+
+.. code-block: shell
+
+   $ git checkout -b feature-branch
+   $ git commit -a -m "Commit messasge."
+   $ git push
+   $ hub pull-request
+
+This process uses GitHub as the review tool and for tracking the
+status of feature branches (in progress, proposed for review, merged).
+
+By allowing pull requests, the GHC project gets to tap into the
+largest social network of programmers in the world, giving it greater
+visibility and allowing contributors to use a familiar workflow that
+has very little overhead (no extra account to create, no additional
+tool to install, no new interface to learn, no unfamiliar commands to
+type).
+
+The objective of this proposal is to make contributing to GHC easier
+through the use of familiar and lightweight tools and platforms, with
+a particular emphasis on encouraging first-time and occasional
+contributors. It is expected that this will have a knock-on on the
+pace of development of the project: more first-time contributors and
+more frequent occasional contributors means more regular contributors
+down the line.
+
+It is often said that any process should instead optimize for regular
+contributors. But the proposed changes below don't imply any changes
+to the current workflow for existing contributors and existing
+reviewers that don't want to use GitHub, so the two are not mutually
+exclusive.
+
+The point is, **how patches should be reviewed is a matter of
+agreement between the contributor and the reviewer(s)**. Reviewers'
+role is simply to vet the patch, something that is done systematically
+in the case of contributors without commit bits. Through which vector
+a patch is accepted isn't significant, provided the reviewers are
+comfortable with it. GHC's release manager or some other role should
+ensure that all potential reviewers are identified, and move the patch
+to Phabricator if need be.
+
+Proposed Change
+---------------
+
+1. Start welcoming PRs on GH.
+2. Lock the ``master`` branch of the ``ghc`` repository on Github to avoid
+   force pushes.
+3. Setup automatic 2-way sync between the GitHub mirror of GHC and the
+   repository in Phabricator.
```

Comment:
Is that easily possible? And is it required? I am happy to accept small Github PRs that I review using plain git commands, fetching from the PR branch that GitHub tells me about and pushing to `git.haskell.org`?